### DISCUSSION TITLE
How would I go about transforming a DataCollection to a typescript Record

### DISCUSSION THREAD

**Discussion by ampersand** [spatie/typescript-transformer#64] [Q&A]
I currently have a Data object like the following.

```php
class ItemData extends Data
{
    /** @var DataCollection<string, SizeData> */
    public DataCollection $sizes;
}
```

I would like to transform it to a typescript type like below.

```ts
export type ItemData = {
    sizes: Record<string, SizeData>
}
```

However, it gets transformed to `Array<SizeData>` instead.  The only method I've found to transform it to a `Record` is to change my PHP type to an array and use the `RecordTypeScriptType`.  

```php
class ItemData extends Data
{
    #[RecordTypeScriptType('string', SizeData::class)]
    public array $sizes;
}
```

With this approach I lose the features of a `DataCollection`.

Does anyone have suggestions on how I  can use a `DataCollection` in my PHP object and also transform it to a TypeScript `Record`?

**Comment by rubenvanassche** [comment]
You can use both of them:

```php
class ItemData extends Data
{
    #[RecordTypeScriptType('string', SizeData::class)]
    /** @var DataCollection<string, SizeData> */
    public array $sizes;
}
```

Not the most beautiful but the best solution I'm afraid.
